generic_config_updater/gu_common.py

```diff
@@ -174,3 +188,562 @@ def convert_sonic_yang_patch_to_config_db_patch(self, patch):
         target_config_db = self.config_wrapper.convert_sonic_yang_to_config_db(target_yang)
 
         return self.generate_patch(current_config_db, target_config_db)
+
+class OperationType(Enum):
+    ADD = 1
+    REMOVE = 2
+    REPLACE = 3
+
+class OperationWrapper:
+    OP_KEYWORD = "op"
+    PATH_KEYWORD = "path"
+    VALUE_KEYWORD = "value"
+
+    def create(self, operation_type, path, value=None):
+        op_type = operation_type.name.lower()
+
+        operation = {OperationWrapper.OP_KEYWORD: op_type, OperationWrapper.PATH_KEYWORD: path}
+
+        if operation_type in [OperationType.ADD, OperationType.REPLACE]:
+            operation[OperationWrapper.VALUE_KEYWORD] = value
+
+        return operation
+
+class PathAddressing:
+    """
+    Path refers to the 'path' in JsonPatch operations: https://tools.ietf.org/html/rfc6902
+    The path corresponds to JsonPointer: https://tools.ietf.org/html/rfc6901
+
+    All xpath operations in this class are only relevent to ConfigDb and the conversion to YANG xpath.
+    It is not meant to support all the xpath functionalities, just the ones relevent to ConfigDb/YANG.
+    """
+    PATH_SEPARATOR = "/"
+    XPATH_SEPARATOR = "/"
+    def get_path_tokens(self, path):
+        # TODO: convert int tokens to int
+        tokens = path.split(PathAddressing.PATH_SEPARATOR)[1:]
+
+        # Because the characters '~' (%x7E) and '/' (%x2F) have special
+        # meanings in JSON Pointer, '~' needs to be encoded as '~0' and '/'
```

Comment:
>JSON Pointer [](http://example.com/codeflow?start=22&length=12)

Did you try [https://pypi.org/project/jsonpointer/](https://pypi.org/project/jsonpointer/) ?

Reply:
still looking into this

Reply:
updated